### PR TITLE
doc: introduce documentation about the unified topology design

### PR DESCRIPTION
**Pull request by mbroadst**
## Description
This begins the process of documenting the general design of the Unified Topology. Hopefully this should serve as a good resource to point confused users to, as well as guidance for our continued development of this piece of the driver.

### CODE REVIEW

File: docs/reference/content/reference/unified-topology/index.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  - remove confusing functionality which could be potentially dangerous for our users
+
+## How to use it
+
+The unified topology is available now behind the `useUnifiedTopology` feature flag. You can opt in to using it by passing the option to your `MongoClient` constructor:
+
+```js
+const client = MongoClient('mongodb://localhost:27017', { useUnifiedTopology: true });
+```
+
```
Comment (daprahamian) [inline docs/reference/content/reference/unified-topology/index.md]: We should add:

1. Eventually, in later minor versions, `useUnifiedTopology` will default to `true`
2. In the next major version, the unified topology will be the default, and we will remove the legacy topology behavior.

File: docs/reference/content/reference/unified-topology/index.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+const coll = client.db('test').collection('foo');
+await coll.insert({ test: 'document' });
+const docs = coll.find({ test: 1 }, { readPreference: 'secondary' }).toArray();
+console.dir({ docs });
+await client.close();
+```
+
+A default `ReadPreference` of "primary" is used for the first write, and a part of awaiting that insert involves initiating connections to all servers in a cluster, selecting a server and executing the operation. Errors will surface at the callsite of any given operation, giving the user more fine-grained control over error handling.
+
+#### Why does `MongoClient.isConnected` always return `true`?
```
Comment (daprahamian) [inline docs/reference/content/reference/unified-topology/index.md]: Probably add a note that `isConnected` will be superfluous soon / go away?

File: docs/reference/content/reference/unified-topology/index.md (hunk 1 above)
Comment (mbroadst) [reply docs/reference/content/reference/unified-topology/index.md]: added a sentence for this

File: docs/reference/content/reference/unified-topology/index.md (hunk 2 above)
Comment (mbroadst) [reply docs/reference/content/reference/unified-topology/index.md]: added a note for this

**Comment by mbroadst** [comment]
@daprahamian yes, I'll convert it to RST. Should we remove the markdown folders for docs completely now that they are converted to RST?

**Pull request merged**